srcs/flipclock.c

```diff
@@ -194,6 +194,60 @@ int _flipclock_parse_key_value(char *line, char **key, char **value)
 	return 0;
 }
 
+int _flipclock_parse_hex_char(char c)
+{
+	if (c >= 'a' && c <= 'f') {
+		return c - 'a' + 10;
+	} else if (c >= 'A' && c <= 'F') {
+		return c - 'A' + 10;
+	} else if (c >= '0' && c <= '9') {
+		return c - '0';
+	} else {
+		return -1;
+	}
+}
+
+int _flipclock_parse_color(const char *value, SDL_Color *color)
+{
+	const unsigned long value_str_len = strlen(value);
+	if (value_str_len == 0) {
+		LOG_ERROR("No color specified!\n");
+		return -1;
+	} else if (value[0] != '#') {
+		LOG_ERROR("Color must start with `#`!\n");
+		return -2;
+	} else if (value_str_len == 7 || value_str_len == 9) {
+		bool is_all_hexcode = true;
+		for (int i = 1; i < value_str_len; i++) {
+			if (!isxdigit(value[i])) {
+				is_all_hexcode = false;
+				break;
+			}
+		}
+		if (is_all_hexcode) {
+			color->r = (_flipclock_parse_hex_char(value[1]) * 16)
+				+ _flipclock_parse_hex_char(value[2]);
```

Comment:
What will happen if `-1` is returned here? We may use some bit operation and `atoi`?

Reply:
Sorry, `strtol` is better.